core/src/main/java/com/nightscout/core/drivers/AbstractDevice.java

```diff
@@ -0,0 +1,30 @@
+package com.nightscout.core.drivers;
+
+import com.nightscout.core.model.DownloadResults;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+abstract public class AbstractDevice {
+    protected final Logger log = LoggerFactory.getLogger(this.getClass());
+    DeviceTransport transport;
+
+    public AbstractDevice(DeviceTransport transport) {
```

Comment:
It is more that every device that we can connect to has some sort of transport mechanism that it will use - be it usb/serial or bluetooth.  I can move it, but I'm pretty sure most of the devices we want to connect to will need it. I also see this class as filling out a little more as we discover more about the devices we connect to.